scalding-core/src/main/scala/com/twitter/scalding/typed/TypedPipe.scala

```diff
@@ -115,6 +118,7 @@ object TypedPipe extends Serializable {
  */
 trait TypedPipe[+T] extends Serializable {
 
+  private def primitiveOrderedBufferSupplier[T]: OrderedSerialization[T] = macro impl.OrderedSerializationProviderImpl[T]
```

Comment:
what is `primitive` about this? Also, can we stop using the `OrderedBuffer` name which is the old name before `OrderedSerialization`. Why not `orderedSerialialization[T]`